### PR TITLE
Add global guassian parents without `Flint` in name

### PR DESCRIPTION
**Pull request by lgoettgens**
As discussed in https://github.com/Nemocas/Nemo.jl/pull/1935#issuecomment-2459500735

Once this is merged, I'll update https://github.com/Nemocas/Nemo.jl/pull/1935, which will then reduce to removing the 4 constants definition in `Nemo.jl` and the corresponding exports.

### CODE REVIEW

File: src/gaussiannumbers/GaussianNumberTypes.jl [hunk 1]
```diff
@@ -3,7 +3,8 @@
 struct ZZiRing <: Ring
 end
 
-const FlintZZi = ZZiRing()
+const ZZi = ZZiRing()
```
Comment (fingolfin) [inline src/gaussiannumbers/GaussianNumberTypes.jl:6]: Why do we need these globals at all?

Comment (lgoettgens) [reply src/gaussiannumbers/GaussianNumberTypes.jl:6]: per request of @thofma in https://github.com/Nemocas/Nemo.jl/pull/1935#issuecomment-2459431082

File: src/gaussiannumbers/GaussianNumberTypes.jl [hunk 2]
```diff
@@ -3,7 +3,8 @@
...
+GaussianIntegers() = ZZi
```
Comment (fingolfin) [inline src/gaussiannumbers/GaussianNumberTypes.jl:7]: I'd be tempted to remove this, too, but it is mentioned in the docs *sigh*. But perhaps we can at least rename it to `gaussian_integers` to fit our naming conventions (I guess we need to keep the old name around for a bit, too)

**Review by fingolfin — APPROVED** [review]
Fine by me as long as we don't export QQi and ZZi

**Pull request merged**